## Entity Data Science Program Final Project

This is for Entity Data Science Program Final Project. We are analyzing diamonds data set to figure out if there is a relationship between clarity/quality of diamond and price of diamond with controlling color of diamond and how depth of diamond impacts the carat of diamond. 

As it can be seen there two part ofs analysis. Answering the first question, we are going to perform ANCOVA(Analysis of Covariance) to figure out the relationship between clarity/quality of diamond and price of diamond with controlling color of diamond. Our independent and dependent variables are:

Independent Variables:

                      Clarity: Categorical variable with 8 levels
                      Color: Categorical variable with 7 levels

Dependent Variable:
                      
                      Price: Continuous variable
                      
                      
Second part of the analysis, we are going to investigate how depth of diamond impacts the carat of diamond and we are going to perform lineer regression. Our variables are:

Independent Variable:

                      Depth: Continuous variable
                      
Dependent Variable:

                      Carat: Continuous variable



**Importing Libraries:**

In [2]:
import numpy as np
import pandas as pd
from pingouin import ancova #pip install pingouin before importing the library.
from sklearn.linear_model import LinearRegression

**Loading Data:**

In [4]:
diamond=pd.read_csv("/Users/nukhet/Entity_Final_Project/Diamonds.csv")

**Exploring Data**

In [5]:
diamond.head()

Unnamed: 0  carat      cut color clarity  depth  table  price     x     y  \
0           0   0.23    Ideal     E     SI2   61.5   55.0    326  3.95  3.98   
1           1   0.21  Premium     E     SI1   59.8   61.0    326  3.89  3.84   
2           2   0.23     Good     E     VS1   56.9   65.0    327  4.05  4.07   
3           3   0.29  Premium     I     VS2   62.4   58.0    334  4.20  4.23   
4           4   0.31     Good     J     SI2   63.3   58.0    335  4.34  4.35   

      z  
0  2.43  
1  2.31  
2  2.31  
3  2.63  
4  2.75

In [6]:
diamond.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 53940 entries, 0 to 53939
Data columns (total 11 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  53940 non-null  int64  
 1   carat       53940 non-null  float64
 2   cut         53940 non-null  object 
 3   color       53940 non-null  object 
 4   clarity     53940 non-null  object 
 5   depth       53940 non-null  float64
 6   table       53940 non-null  float64
 7   price       53940 non-null  int64  
 8   x           53940 non-null  float64
 9   y           53940 non-null  float64
 10  z           53940 non-null  float64
dtypes: float64(6), int64(2), object(3)
memory usage: 4.5+ MB


In [8]:
diamond.clarity.unique()

array(['SI2', 'SI1', 'VS1', 'VS2', 'VVS2', 'VVS1', 'I1', 'IF'],
      dtype=object)

In [9]:
diamond.color.unique()

array(['E', 'I', 'J', 'H', 'F', 'G', 'D'], dtype=object)

In [10]:
diamond.price.min()

326

In [11]:
diamond.price.max()

18823

In [12]:
diamond.depth.min()

43.0

In [13]:
diamond.depth.max()

79.0

In [14]:
diamond.carat.min()

0.2

In [15]:
diamond.carat.max()

5.01

**Carat Weight:** 0.2 - 5.01

**Price in US dollars:** 326 - 18823

**Color:** from J (Worst) to D (Best)

**Clarity:** I1 (Worst), SI2, SI1, VS2, VS1, VVS2, VVS1, IF (Best)

**Depth:** 43 - 79

 Depth is calculated by total depth percentage = z / mean(x, y) = 2 * z / (x + y)
 
 x length in mm

 y width in mm 

 z depth in mm 



**Data Wrangling**

Let's count how many missing values how many missing values each columns have.

In [20]:
diamond.isna().sum()

Unnamed: 0    0
carat         0
cut           0
color         0
clarity       0
depth         0
table         0
price         0
x             0
y             0
z             0
dtype: int64

As it can be seen data is clean which means there is no missing values(NA's) on diamond data.

For ANCOVA 